presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.util;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.Math.max;
+
+public class StringTableUtils
+{
+    public static String getShortestTableStringFormat(List<List<String>> table)
+            throws IllegalArgumentException
+    {
+        if (table.size() == 0) {
+            return "";
+        }
+
+        int tableWidth = table.get(0).size();
+        int[] lengthTracker = new int[tableWidth];
+        for (List<String> row : table) {
+            if (row.size() != lengthTracker.length) {
+                throw new IllegalArgumentException("all lists in table must be the same size");
```

Comment:
format(`all rows in the table are expected to have exactly same number of columns: %s != %s`, tableWidth, row.size())